docs/reference/remap/errors/650_chained_comparison_operators.cue

```diff
@@ -0,0 +1,39 @@
+package metadata
+
+remap: errors: "650": {
+	title: "Chained comparison operators"
+
+	description: """
+		You've chained multiple comparison operators together in a way that can't result in a valid expression.
+		"""
+
+	rationale: """
+		Comparison operators can only operate on two operands, e.g. `1 != 2`. Chaining them together, as in
+		`1 != 2 != 3`, produces a meaningless non-expression.
+		"""
+
+	resolution: """
+		Use comparison operators only on a left-hand- and a right-hand-side value.
```

Comment:
It's worth pointing out that you can still group operations together to get the desired result:

```
> (1 == 1) == 1
false
```

The reason we have this check in place is to avoid people expecting this:

```
2 < 3 < 4
```

to mean "3 is greater than 2, and smaller than 4", which is not what it means, instead it means:

```
(2 < 3) < 4
```

Or: "true is smaller than 4", which would result in a compile-time error, obviously.